sympy/combinatorics/perm_groups.py

```diff
@@ -3815,7 +3815,301 @@ def _sylow_reduce(mu, nu):
+            k_gens = K.stabilizer(b).generators
+            for y in k_gens:
+                new_rel = transversal[b]
+                gens = K.generator_product(y, original=True)
+                for g in gens[::-1]:
+                    new_rel = new_rel*phi.invert(g)
+                new_rel = new_rel*transversal[b]**-1
+
+                perm = phi(new_rel)
+                try:
+                    gens = K.generator_product(perm, original=True)
+                except ValueError:
+                    return False, perm
+                for g in gens:
+                    new_rel = new_rel*phi.invert(g)**-1
+                if new_rel not in rels:
+                    rels.append(new_rel)
+
+        for gamma in gammas:
+            new_rel = transversal[gamma]*phi.invert(z)*transversal[gamma^z]**-1
+            perm = phi(new_rel)
+            try:
+                gens = K.generator_product(perm, original=True)
+            except ValueError:
+                return False, perm
+            for g in gens:
+               new_rel = new_rel*phi.invert(g)**-1
+            if new_rel not in rels:
+                rels.append(new_rel)
+
+        return True, rels
+
+    def strong_presentation(G):
+        '''
+        Return a strong finite presentation of `G`. The generators
+        of the returned group are in the same order as the strong
+        generators of `G`.
+
+        The algorithm is based on Sims' Verify algorithm described
+        in [1], Chapter 6.
+
+        Example
+        =======
+        >>> from sympy.combinatorics.perm_groups import PermutationGroup
+        >>> from sympy.combinatorics.named_groups import DihedralGroup
+        >>> P = DihedralGroup(4)
+        >>> G = P.strong_presentation()
+        >>> P.order() == G.order()
+        True
+
+        See Also
+        ========
+        presentation, _verify
+
+        '''
+        from sympy.combinatorics.fp_groups import (FpGroup,
+                                            simplify_presentation)
+        from sympy.combinatorics.free_groups import free_group
+        from sympy.combinatorics.homomorphisms import (block_homomorphism,
+                                           homomorphism, GroupHomomorphism)
+        from itertools import product
+
+        strong_gens = G.strong_gens[:]
+        stabs = G.basic_stabilizers[:]
+        base = G.base[:-1]
+
+        # injection from a free group on len(strong_gens)
+        # generators into G
+        gen_syms = [('x_%d'%i) for i in range(len(strong_gens))]
+        F = free_group(', '.join(gen_syms))[0]
+        phi = homomorphism(F, G, F.generators, strong_gens)
+
+        H = stabs.pop()
+        if not stabs:
+            # there is only one stabilizer so a strong presentation
+            # is the same as
+            return H.presentation(eliminate_gens=False)
+        # H is a group on one generator and this is the relator for it
+        rels = [F.generators[-1]**H.generators[0].order()]
```

Comment:
I have not succeeded in proving this. What would happen with Klein's 4-group?

Reply:
It's not the case in general but I was under a strong impression that `schreier_sims` orders the base in a way that this always happens because somehow this was the case with all of the random tests I ran (which was many). But turns out it was a wrong impression (and also silly of me to not verify, sorry about that).

Klein's 4-group always has a base of length 1 so there is only one stabilizer and it doesn't get to this point in the algorithm. But if you throw in a disjoint permutation into the generating set and order the generators in a specific way, it gives a counterexample. E.g.
```
>>> a = Permutation(0, 1)(2, 3)
>>> b = Permutation(0, 2)(3, 1)
>>> c = Permutation(4, 5)
>>> V = PermutationGroup(c, a, b)
```
(but not `PermutationGroup(a, b, c)`). 

I'll  make sure that if there are more than one generators, the stabilizer is completed by the main for loop. Also, should probably add it to the tests.